client/lib/load-script/index.js

```diff
@@ -3,92 +3,186 @@
+	}
+}
+
+// NOTE: __config and __loadScript are used for testing only.
+export function loadjQueryDependentScript(
+	url,
+	callback,
+	__config = config,
+	__loadScript = loadScript,
+) {
+	debug( `Loading a jQuery dependent script from "${ url }"` );
 
-var loadjQueryDependentScript = function( scriptURL, callback ) {
 	// It is not possible to expose jQuery globally in Electron App: https://github.com/atom/electron/issues/254.
 	// It needs to be loaded using require and npm package.
-	if ( config.isEnabled( 'desktop' ) ) {
+	if ( __config.isEnabled( 'desktop' ) ) {
 		window.$ = window.jQuery = require( 'jquery' );
 	}
+
 	if ( window.jQuery ) {
-		loadScript( scriptURL, callback );
+		__loadScript( url, callback );
 		return;
 	}
-	loadScript( JQUERY_URL, function( error ) {
+
+	__loadScript( JQUERY_URL, function( error ) {
 		if ( error ) {
 			callback( error );
 		}
-		loadScript( scriptURL, callback );
+		__loadScript( url, callback );
 	} );
-};
+}
 
-const removeScriptCallback = function( url, callback ) {
-	if ( ! callbacksForURLsInProgress[ url ] ) {
-		return;
-	}
+//
+// Internal functions
+//
 
-	const index = callbacksForURLsInProgress[ url ].indexOf( callback );
+export function exposeInternalMethodsForTesting() {
+	return {
+		_getCallbacks,
+		_addScriptCallback,
+		_executeCallbacks,
+		_createScriptElement,
+		_attachToHead,
+		_handleRequestSuccess,
+		_handleRequestError,
+	};
+}
 
-	if ( -1 === index ) {
-		return;
+function _getCallbacks() {
+	return callbacksForURLsInProgress;
+}
+
+function _addScriptCallback( url, callback ) {
+	if ( isLoading( url ) ) {
+		debug( `Adding a callback for an existing script from "${ url }"` );
+		callbacksForURLsInProgress[ url ].push( callback );
+	} else {
+		debug( `Adding a callback for a new script from "${ url }"` );
+		callbacksForURLsInProgress[ url ] = [ callback ];
 	}
+}
 
-	if ( 1 === callbacksForURLsInProgress[ url ].length ) {
+function _executeCallbacks( url, callbackArguments = null ) {
+	if ( callbacksForURLsInProgress.hasOwnProperty( url ) ) {
+		const debugMessage = `Executing callbacks for "${ url }"`;
+		debug( callbackArguments === null ? debugMessage : debugMessage + ` with args "${ callbackArguments }"` );
+
+		callbacksForURLsInProgress[ url ].filter( cb => typeof cb === 'function' ).forEach( function( cb ) {
+			cb( callbackArguments );
+		} );
 		delete callbacksForURLsInProgress[ url ];
-		return;
 	}
+}
 
-	callbacksForURLsInProgress[ url ].splice( index, 1 );
-};
-
-module.exports = {
-	loadScript: loadScript,
-	loadjQueryDependentScript: loadjQueryDependentScript,
-	removeScriptCallback: removeScriptCallback,
-	JQUERY_URL: JQUERY_URL
-};
+function _createScriptElement( url, onload = _handleRequestSuccess, onerror = _handleRequestError ) {
+	debug( `Creating script element for "${ url }"` );
+	const script = document.createElement( 'script' );
+	script.src = url;
+	script.type = 'text/javascript';
+	script.async = true;
+	script.onload = () => {
```

Comment:
This is just a nitpick, since I don't think we call this a lot ( correct me if got it wrong and we actually use it alot )
For each call of `_createScriptElement` this we create:
1 onload lambda function
1 onerror lambda function
1 copy of _handleRequestSuccess
1 copy of _handleRequestError


while we could do something like:

```js
scriptElementToUrlMap.set( script, url );
script.onload = _handleRequestSuccess;

function _handleRequestSuccess( ev, __executeCallbacks = _executeCallbacks ) {
    const url = scriptElementToUrlMap.get( ev.target );
	debug( `Handling successful request for "${ url }"` );
	__executeCallbacks( url );
	ev.target.onload = null;
}

```